qt/scientific_interfaces/Muon/ALCDataLoadingPresenter.cpp

```diff
@@ -463,5 +385,118 @@ bool ALCDataLoadingPresenter::isLoading() const { return m_loadingData; }
  */
 void ALCDataLoadingPresenter::cancelLoading() const { m_LoadingAlg->cancel(); }
 
+void ALCDataLoadingPresenter::handleInstrumentChanged(std::string instrument) {
+  // Clear path as instrument has changed
+  m_view->setPath(std::string{});
+
+  // Update instrument
+  m_view->setInstrument(instrument);
+
+  // User cannot load yet as path not set
+  m_view->enableLoad(false);
+
+  // Turn off auto add
+  m_view->enableRunsAutoAdd(false);
+  m_view->toggleRunsAutoAdd(false);
+}
+
+void ALCDataLoadingPresenter::handleManageDirectories() {
+  MantidQt::API::ManageUserDirectories::openManageUserDirectories();
+}
+
+/**
+ * The watched directory has been changed - update flag.
+ * @param path :: [input] Path to directory modified (not used)
+ */
+void ALCDataLoadingPresenter::updateDirectoryChangedFlag(const QString &path) {
+  Q_UNUSED(path); // just set the flag, don't need the path
+  m_directoryChanged = true;
+}
+
+void ALCDataLoadingPresenter::startWatching(bool watch) {
+  if (watch) {
+    // Get path to watch and add to watcher
+    const auto path = m_view->getPath();
+    m_watcher.addPath(QString::fromStdString(path));
+
+    // start a timer that executes every second
+    m_timerID = startTimer(1000);
+  } else {
+    // Check if watcher has a directory, then remove all
+    if (!m_watcher.directories().empty()) {
+      m_watcher.removePaths(m_watcher.directories());
+    }
+
+    // Stop timer
+    killTimer(m_timerID);
+
+    // Reset latest auto run number and was range
+    m_lastRunLoadedAuto = -2;
```

Comment:
why -2?

Reply:
I can see how this is not clear I will add a comment. 
The reason for -2 is this part of the code: https://github.com/mantidproject/mantid/blob/f955831bbfea204ad849269549c57cfbb18fbafd/qt/scientific_interfaces/Muon/ALCDataLoadingPresenter.cpp#L475

By resetting m_lastRunLoadedAuto to -2 we ensure that when you add 1 to it you never get a valid run number i.e. it is still negative.